tests/test_util.py

```diff
@@ -0,0 +1,134 @@
+import os
+import types
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    # `Python 2` or lower than `Python 3.3` does not
+    # have the `unittest.mock` module built-in
+    import mock
+from pythonforandroid import util
+
+
+class TestUtil(unittest.TestCase):
+    @mock.patch("pythonforandroid.util.makedirs")
+    def test_ensure_dir(self, mock_makedirs):
+        util.ensure_dir("fake_directory")
+        mock_makedirs.assert_called_once_with("fake_directory")
+
+    @mock.patch("shutil.rmtree")
+    @mock.patch("pythonforandroid.util.mkdtemp")
+    def test_temp_directory(self, mock_mkdtemp, mock_shutil_rmtree):
+        mock_mkdtemp.return_value = "/temp/any_directory"
+        with util.temp_directory():
+            mock_mkdtemp.assert_called_once()
+        mock_shutil_rmtree.assert_called_once_with("/temp/any_directory")
+
+    @mock.patch("pythonforandroid.util.chdir")
+    def test_current_directory(self, moch_chdir):
+        chdir_dir = "/temp/any_directory"
+        # test chdir to existing directory
+        with util.current_directory(chdir_dir):
+            moch_chdir.assert_called_once_with("/temp/any_directory")
+        moch_chdir.assert_has_calls(
+            [
+                mock.call("/temp/any_directory"),
+                mock.call(os.getcwd()),
+            ]
+        )
+
+    def test_current_directory_exception(self):
+        # test chdir to non-existing directory, should raise error
+        # for py3 the exception is FileNotFoundError and IOError for py2, to
+        # avoid introduce conditions, we test with a more generic exception
+        with self.assertRaises(OSError):
+            with util.current_directory("/fake/directory"):
+                # the line below will never be executed
```

Comment:
If it will never be executed, why not prove it by raising an error? But this is a super minor point, I have no issue with the test.

Reply:
You could use a `pass`, but I think it's also fine as it is. Also do you know you can chain context managers without indenting multiple `with`, sometimes it's readable enough:
```python
with self.assertRaises(OSError), util.current_directory("/fake/directory"):
  pass
```